src/compiler-src.md

```diff
@@ -101,12 +93,9 @@ At the very bottom of the dependency tree are a handful of crates that are used
 by the whole compiler (e.g. [`rustc_span`]). The very early parts of the
 compilation process (e.g. parsing and the AST) depend on only these.
 
-Pretty soon after the AST is constructed, the compiler's [query system][query]
-gets set up.  The query system is set up in a clever way using function
-pointers. This allows us to break dependencies between crates, allowing more
-parallel compilation.
-
-However, since the query system is defined in [`rustc_middle`], nearly all
+After the AST is constructed, the compiler's [query system][query]
+gets set up.[^query-setup]
```

Comment:
IMO too many footnotes is more distracting than just having the text inline.

Reply:
Too many footnotes is certainly distracting! In this case, I didn't think this implementation detail belongs in the main body text of a "Big picture" section.

Reply:
I agree in some ways, but the implementation detail informs the code structure, so it still seems potentially like it belongs.